### PR TITLE
Allow passing through of (almost) all params available …

### PR DESCRIPTION
**Pull request by stefanhorning**
…on boto3 methods for aws_api_gateway.

##### SUMMARY
The boto methods in use for aws_api_gateway allow more params. So this PR allows to set them through the Ansible module too. See https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/apigateway.html#client

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
aws_api_gateway module.

##### ADDITIONAL INFORMATION

Simple extension of module params.

### CODE REVIEW

File: lib/ansible/modules/cloud/amazon/aws_api_gateway.py [hunk 1]
```diff
@@ -271,9 +320,31 @@ def configure_api(client, api_data=None, api_id=None, mode="overwrite"):
 
 
 @AWSRetry.backoff(**retry_params)
-def create_deployment(client, api_id=None, stage=None, description=None):
-    # we can also get None as an argument so we don't do this as a default
-    return client.create_deployment(restApiId=api_id, stageName=stage, description=description)
+def create_deployment(client, api_id=None, stage=None, description=None, cache_enabled=False, cache_size='0.5',
+        variables=None, canary_settings=None, tracing_enabled=False):
```
Comment (stefanhorning) [inline lib/ansible/modules/cloud/amazon/aws_api_gateway.py]: This method should probably be refactored to use keyword arguments instead, to reduce params list here.

**Comment by stefanhorning** [comment]
@mikedlr @willthames please review.

**Comment by stefanhorning** [comment]
Rebased on latest devel to make merging easier.

**Comment by jillr** [comment]
Hi @stefanhorning, apologies about the delay on this PR.  Could you please add a changelog for the new options?

File: lib/ansible/modules/cloud/amazon/aws_api_gateway.py [hunk 2]
```diff
@@ -69,6 +68,43 @@
         AWS console.
     default: Automatic deployment by Ansible.
     type: str
+  cache_enabled:
+    description:
+      - Enable API GW caching of backend responses. Defaults to false
```
Comment (jillr) [inline lib/ansible/modules/cloud/amazon/aws_api_gateway.py]:
```suggestion
      - Enable API GW caching of backend responses. Defaults to false.
```

File: lib/ansible/modules/cloud/amazon/aws_api_gateway.py [hunk 3]
```diff
@@ -69,6 +68,43 @@
...
+    default: '0.5'
+    version_added: '2.10'
+  stage_variables:
+    description:
+      - ENV variables for the stage. Define a dict of key values pairs for variables.
+    type: dict
+    version_added: '2.10'
+  stage_canary_settings:
+    description:
+      - Canary settings for the deployment of the stage
```
Comment (jillr) [inline lib/ansible/modules/cloud/amazon/aws_api_gateway.py]:
```suggestion
      - Canary settings for the deployment of the stage.
```

File: lib/ansible/modules/cloud/amazon/aws_api_gateway.py [hunk 4]
```diff
@@ -98,20 +134,32 @@
     swagger_file: my_api.yml
     stage: production
     deploy_desc: Make auth fix available.
+    cache_enabled: true
+    cache_size: '1.6'
+    tracing_enabled: true
```
Comment (jillr) [inline lib/ansible/modules/cloud/amazon/aws_api_gateway.py]: Could you please add examples for the other new options?

File: lib/ansible/modules/cloud/amazon/aws_api_gateway.py [hunk 5]
```diff
@@ -98,20 +134,32 @@
...
+    returned: success
+    type: str
+    sample: '0ln4zq7p86'
+configure_response:
+    description: AWS response from the API configure call
+    returned: success
+    type: dict
+    sample: { api_key_source: "HEADER", created_at: "2020-01-01T11:37:59+00:00", id: "0ln4zq7p86" }
+deploy_response:
+    description: AWS responce from the API deploy call
```
Comment (jillr) [inline lib/ansible/modules/cloud/amazon/aws_api_gateway.py]:
```suggestion
    description: AWS response from the API deploy call
```

File: lib/ansible/modules/cloud/amazon/aws_api_gateway.py [hunk 6]
```diff
@@ -69,6 +68,43 @@
...
+  tracing_enabled:
+    description:
+      - Specifies whether active tracing with X-ray is enabled for the API GW stage.
+    type: bool
+    version_added: '2.10'
+  endpoint_type:
+    description:
+      - Type of endpoint configuration, use EDGE for an edge optomized API endpoint,
+      - REGIONAL for just a regional deploy or PRIVATE for a private API.
+      - This will flag will only be used when creating a new API Gateway setup, not for updates.
```
Comment (jillr) [inline lib/ansible/modules/cloud/amazon/aws_api_gateway.py]:
```suggestion
      - Type of endpoint configuration, use C(EDGE) for an edge optimized API endpoint,
      - C(REGIONAL) for just a regional deploy or C(PRIVATE) for a private API.
      - This flag will only be used when creating a new API Gateway setup, not for updates.
```

**Comment by stefanhorning** [comment]
@jillr no problem. Added a changelog fragment now, hope it's ok.

**Review by jillr — COMMENTED** [review]
Couple of docs nits

**Comment by stefanhorning** [comment]
Thanks for spotting those typos. Now corrected all of them and also imroved the docs a bit.

**Review by jillr — APPROVED** [review]
LGTM - thanks @stefanhorning

**Pull request merged**